database/migrations/1749227968774_alter_field_of_studies_and_add_study_types_table.ts

```diff
@@ -0,0 +1,71 @@
+import { BaseSchema } from "@adonisjs/lucid/schema";
+
+export default class extends BaseSchema {
+  protected tableName = "fields_of_studies";
+  protected studiesTypes = ["1DEGREE", "2DEGREE", "LONG_CYCLE"];
+
+  async up() {
+    await this.schema.raw(`
+      DO $$ BEGIN
+        IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'studies_type') THEN
+          CREATE TYPE "studies_type" AS ENUM (${this.studiesTypes.map((type) => `'${type}'`).join(", ")});
+        END IF;
+      END $$;
+      DO $$ BEGIN
+        IF NOT EXISTS (
+            SELECT 1
+            FROM information_schema.columns
+            WHERE table_name = '${this.tableName}' AND column_name = 'studies_type'
+        ) THEN ALTER TABLE "${this.tableName}" ADD COLUMN "studies_type" "studies_type" NULL;
+        END IF;
+      END $$;
+    `);
+    await this.db.rawQuery(`
+      UPDATE ${this.tableName}
+      SET studies_type = CASE
+                   WHEN semester_count = 12 THEN 'LONG_CYCLE'::studies_type
+                   WHEN is_2nd_degree = true THEN '2DEGREE'::studies_type
+                   ELSE '1DEGREE'::studies_type
+        END
+    `);
+    this.schema.alterTable(this.tableName, (table) => {
+      table.dropNullable("studies_type");
+      table.dropColumn("is_2nd_degree");
+      table.dropColumn("semester_count");
+    });
+  }
+
+  async down() {
+    await this.schema.raw(`
+      DO $$ BEGIN
+        IF NOT EXISTS (
+            SELECT 1
+            FROM information_schema.columns
+            WHERE table_name = '${this.tableName}' AND column_name = 'is_2nd_degree'
+        ) THEN ALTER TABLE "${this.tableName}" ADD COLUMN "is_2nd_degree" boolean NULL;
+        END IF;
+        IF NOT EXISTS (
+            SELECT 1
+            FROM information_schema.columns
+            WHERE table_name = '${this.tableName}' AND column_name = 'semester_count'
+        ) THEN ALTER TABLE "${this.tableName}" ADD COLUMN "semester_count" INTEGER NULL;
+        END IF;
+      END $$;
+    `);
```

Comment:
same as above - we don't need these checks, just use standard adonis schema builders